backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EditingContextPersistenceService.java

```diff
@@ -67,14 +69,20 @@ public EditingContextPersistenceService(IDocumentRepository documentRepository,
     public void persist(UUID projectId, IEditingContext editingContext) {
         long start = System.currentTimeMillis();
 
+        Object domain = editingContext.getDomain();
+        if (domain instanceof EditingDomain) {
+            EditingDomain editingDomain = (EditingDomain) domain;
+            List<DocumentEntity> documentEntities = this.persist(editingDomain);
+            List<Document> documents = documentEntities.stream().map(new DocumentMapper()::toDTO).collect(Collectors.toList());
+            this.applicationEventPublisher.publishEvent(new DocumentsModifiedEvent(projectId, documents));
+        }
         // @formatter:off
         var optionalDocuments = Optional.ofNullable(editingContext)
             .map(IEditingContext::getDomain)
             .filter(EditingDomain.class::isInstance)
             .map(EditingDomain.class::cast)
             .map(this::persist);
         optionalDocuments.ifPresent(documentEntities -> {
```

Comment:
Am I misreading this part, or are we persisting the domain and documents twice now?

Reply:
I knew something was off, I fixed that later. When we are done with the review, I'll bring back the fix here to prevent this.

Reply:
Fixed in https://github.com/eclipse-sirius/sirius-components/pull/250/commits/ba21062b8e52428dc2e054d1892d78e969f9c82d#diff-ff868b8844dfe60a83b0d1a437294f9413ae2bb4bf3e87649083ea90c3d5f88e